src/main/java/com/harriet/shopiify/order/service/OrderService.java

```diff
@@ -0,0 +1,203 @@
+import com.harriet.shopiify.order.dto.OrderItemDTO;
+import com.harriet.shopiify.order.model.Order;
+import com.harriet.shopiify.order.model.OrderItem;
+import com.harriet.shopiify.order.model.OrderItemKey;
+import com.harriet.shopiify.order.repository.OrderItemRepository;
+import com.harriet.shopiify.order.repository.OrderRepository;
+import com.harriet.shopiify.order.repository.OrderStatusRepository;
+import com.harriet.shopiify.order.vo.OrderCreateVO;
+import com.harriet.shopiify.order.vo.OrderItemVO;
+import com.harriet.shopiify.product.model.Product;
+import com.harriet.shopiify.product.repository.ProductRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.weaver.ast.Or;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class OrderService {
+
+    private final ShippingAddressRepository shippingAddressRepository;
+    private final OrderStatusRepository orderStatusRepository;
+    private final ProductRepository productRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Transactional
+    public Long createOrder(OrderCreateVO vo) throws Exception{
+        Long createdOrderId = null;
+        try{
+            createdOrderId = saveOrder(vo);
+            saveOrderItems(vo);
+            deleteCartItems(vo);
+            reduceStockByOrderedQty(vo);
+        } catch(Exception e){
+            throw new Exception(e.getMessage());
+        }
+        return createdOrderId;
+    }
+
+    private Long saveOrder(OrderCreateVO vo) throws Exception{
+        log.info("creating new order");
+        if (vo.getOrderItems().isEmpty()){
+            throw new UnsupportedOperationException("cannot create new order with no order items");
+        }
+        for ( OrderItemVO item : vo.getOrderItems()){
+            log.info("cart id and product id: {}, {} ",vo.getCartId(), item.getProductId());
+            Optional<CartItem> cartItem = cartItemRepository.findById(toCartItemKey(vo.getCartId(), item.getProductId()));
+            Boolean itemInCart = cartItem.isPresent();
+            log.info("itemInCart: {} ", itemInCart);
+            Boolean orderQtySameInCart = itemInCart ? cartItem.get().getQuantity()==item.getQuantity() : false;
+            log.info("cartItem.get().getQuantity() {}",cartItem.get().getQuantity());
+            log.info("item.getQuantity() {}", item.getQuantity());
+            log.info("orderQtySameInCart: {} ", orderQtySameInCart);
+            if (!itemInCart || !orderQtySameInCart){
+                throw new Exception("ordered item does not match cart time for item: " + item.getProductId());
```

Comment:
I think you mean `ordered item does not match cart quantity for item:` here. Also you could throw `IllegalArgumentException` instead of `Exception` to avoid using checked exceptions.

Reply:
same as above